src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -623,10 +623,15 @@ def merge_kubernetes_configurations(existing_file, addition_file):
     with open(addition_file) as stream:
         addition = yaml.load(stream)
 
-    # TODO: this will always add, we should only add if not present
-    existing['clusters'].extend(addition['clusters'])
-    existing['users'].extend(addition['users'])
-    existing['contexts'].extend(addition['contexts'])
+    for i in addition['clusters']:
+        if i not in existing['clusters']:
+            existing['clusters'].extend([i])
```

Comment:
I'm curious, why not use `append` here?  Append one item is equivalent to extend by a list containing one item. Did I miss anything here?